src/main/java/uk/gov/companieshouse/api/accounts/service/impl/CompanyAccountServiceImpl.java

```diff
@@ -85,24 +88,33 @@
         return new ResponseObject<>(ResponseStatus.CREATED, companyAccount);
     }
 
+    @Override
+    public void addLink(String id, LinkType linkType, String link) {
+        CompanyAccountEntity companyAccountEntity = companyAccountRepository.findById(id)
```

Comment:
findById should throw a DataException that we pass up the chain here

Reply:
as discussed, happy for this class to be reworked in a subsequent PR to fit in with the AbstractService interface